Packs/AzureNetworkSecurityGroups/Integrations/AzureNetworkSecurityGroups/AzureNetworkSecurityGroups.py

```diff
@@ -40,7 +40,8 @@ def __init__(self, app_id, subscription_id, resource_group_name, verify, proxy,
                                  # deployed machine, the DEVICE_CODE flow should behave somewhat like a self deployed
                                  # flow and most of the same arguments should be set, as we're !not! using OProxy.
             auth_id=app_id,
-            token_retrieval_url='https://login.microsoftonline.com/organizations/oauth2/v2.0/token',
+            token_retrieval_url='https://login.microsoftonline.com/organizations/oauth2/v2.0/token' if 'Device' in
```

Comment:
2 notes:
1. Why not compare the entire `connection_type` to the correct type - will be more informative and easy to understand
2. If we have another authorization flow, don't we need another token retrieval URL? or None is fine?

Reply:
As for 1, I will update the code as requested.
As for 2, it is fine, because the API module handles it.